### PR TITLE
Test for issue #181

### PR DESCRIPTION
**Pull request by Redominus**
I've created a test to fail the case described in issue #181 .

### CODE REVIEW

**Comment by brendt** [comment]
Thanks! Are you up for submitting a PR to fix the failing test? I'm ok if you added it in this PR.

No problem if you're unable to look at it further, just let me know!

**Comment by Redominus** [comment]
Fixed :smile: .Also pull rebased to lastest tag to keep log more straight.

File: src/LinkAdder.php [hunk 1]
```diff
@@ -88,7 +88,9 @@ protected function normalizeUrl(UriInterface $url): UriInterface
     protected function shouldCrawl(Node $node): bool
     {
         if ($this->crawler->mustRespectRobots()) {
-            return $this->crawler->getRobotsTxt()->allows($node->getValue());
+            if (! $this->crawler->getRobotsTxt()->allows($node->getValue())) {
```
Comment (brendt) [inline src/LinkAdder.php]: Let's write this as one condition, instead of a nested one.

**Comment by brendt** [comment]
Thank you very much! I've added one minor remark

File: src/LinkAdder.php (hunk 1 above)
Comment (Redominus) [reply src/LinkAdder.php]: Done.

**Comment by brendt** [comment]
Thanks! This has been tagged as 4.3.2: https://github.com/spatie/crawler/releases/tag/4.3.2

**Pull request merged**